File: pytools/examples/golang/src/client/proto/req_scene_enter.go
package proto

import (
	"packet"
)

type ReqSceneEnter struct {
	doorId                   uint32
}

func (this *ReqSceneEnter) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint32(this.doorId)

	return pack.Encode(P_REQ_SCENE_ENTER)
}

func (this *ReqSceneEnter) SetDoorId(doorId uint32) {
	this.doorId = doorId
}

File: pytools/examples/lua/client/protocols/ReqSceneEnter.lua
module("ReqSceneEnter", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = ReqSceneEnter})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.door_id = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU32(self.door_id)
	return pack:encode(Msg.P_REQ_SCENE_ENTER)
end


function setDoorId(self, door_id)
	self.door_id = door_id
end

